process/scToProtocol/stakingToPeer.go

```diff
@@ -238,6 +238,11 @@ func (stp *stakingToPeer) updatePeerState(
 		if isNewValidator {
 			account.SetListAndIndex(account.GetShardId(), string(core.NewList), uint32(stakingData.UnJailedNonce))
 		}
+
+		if account.GetList() == string(core.JailedList) {
```

Comment:
This shouldn't be done on else branch, if is not new validator ? Could be the situation when isNewValidator == true &&  account.GetList() == string(core.JailedList) ? In this case account.SetListAndIndex will be executed twice here, first time with core.NewList and second time with core.InactiveList.

Reply:
Agreed. Must add to else branch.

Reply:
No need to change. It is good as it is. If it is a new validator, it will set to new list. Thus it will not get into this branch.

Reply:
ok